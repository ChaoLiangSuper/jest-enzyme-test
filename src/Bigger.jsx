import React from 'react';
import { getBiggerNum } from './functions';

class Bigger extends React.Component {
  render() {
    const { a, b } = this.props;
    return <div>getBiggerNum: {getBiggerNum(a, b)}</div>
  }
}

export default Bigger;
