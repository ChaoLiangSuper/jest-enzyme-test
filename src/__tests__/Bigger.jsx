import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Bigger from '../Bigger';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const propsOfGreatA = {
  a: 2,
  b: 1
}

const propsOfGreatB = {
  a: 1,
  b: 3
}

it('should render default component', () => {
  const component = renderer.create(<Bigger />);
  const snap = component.toJSON();
  expect(snap).toMatchSnapshot();
});

it('should return A with propsOfGreatA', () => {
  const componentA = shallow(<Bigger {...propsOfGreatA} />);
  expect(componentA.html()).toBe('<div>getBiggerNum: 2</div>')
});

it('should return B with propsOfGreatB', () => {
  const componentB = shallow(<Bigger {...propsOfGreatB} />);
  expect(componentB.html()).toBe('<div>getBiggerNum: 3</div>')
});
