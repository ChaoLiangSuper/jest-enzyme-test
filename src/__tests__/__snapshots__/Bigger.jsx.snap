// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default component 1`] = `
<div>
  getBiggerNum: 
</div>
`;
